File: app/src/main/java/com/alex/morgan/bearlist/list/BearListContract.java
package com.alex.morgan.bearlist.list;

import com.alex.morgan.bearlist.Bear;

import java.util.Collection;

interface BearListContract {
    interface View {
        void showBears(Collection<Bear> bears);

        void showLoading(String text);

        void showError();
    }

    interface Presenter {
        void attachView(View view);

        void detachView();

        void onRefreshBears();

        void onShowBearDetail(Bear bear);
    }
}

File: app/src/main/java/com/alex/morgan/bearlist/list/BearListContract.kt
package com.alex.morgan.bearlist.list

import com.alex.morgan.bearlist.Bear

interface BearListContract {
    interface View {
        fun showBears(bears: Collection<Bear>)

        fun showLoading(text: String)

        fun showError()

        fun showMessage(message: String)
    }

    interface Presenter {
        fun attachView(view: View)

        fun detachView()

        fun onRefreshBears()

        fun onShowBearDetail(bear: Bear)

        fun onActivateRubeGoldbergMachine()
    }
}
